src/mimeparser/mimeparser_tests.rs

```diff
@@ -2011,3 +1987,48 @@ async fn test_protected_date() -> Result<()> {
     assert_eq!(alice_msg.get_timestamp(), bob_msg.get_timestamp());
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_huge_image_becomes_file() {
+    let t = TestContext::new_alice().await;
+    receive_imf(
+        &t.ctx,
+        include_bytes!("../../test-data/message/image_huge_36M.eml"),
+        false,
+    )
+    .await
+    .unwrap();
+    let chats = Chatlist::try_load(&t.ctx, 0, None, None).await.unwrap();
```

Comment:
No need to use `Chatlist` -- `receive_imf()` returns `ReceivedMsg` with `msg_ids[0]` as a message id.